tools/system-health-check/health_check.py

```diff
@@ -0,0 +1,210 @@
+        )
+      )
+    ) by (pod)
+  )
+'''
+
+def pod_check():
+    failed = []
+    response = requests.get(prometheus + '/api/v1/query?', params={'query': pods_health_query})
+    response_json = json.loads(response.text)
+    if len(response_json['data']['result']) < 1:
+        return failed
+    else:
+        for result in response_json['data']['result']:
+            failed.append(str(result['metric']['deployment']))
+        return failed
+
+def pvc_check():
+    failed = []
+    response = requests.get(prometheus + '/api/v1/query?', params={'query': pvc_health_query})
+    response_json = json.loads(response.text)
+    if len(response_json['data']['result']) < 1:
+        return failed
+    else:
+        for result in response_json['data']['result']:
+            failed.append("PVC - "+str(result['metric']['persistentvolumeclaim'])+" has less than 200MB available space")
+        return failed
+
+def node_check():
+    failed = []
+    response = requests.get(prometheus + '/api/v1/query?', params={'query': node_health_query})
+    response_json = json.loads(response.text)
+    if len(response_json['data']['result']) < 1:
+        return failed
+    else:
+        for result in response_json['data']['result']:
+            failed.append("Node - "+ result['metric']['instance'] + " is not healthy")
+        return failed
+
+def slo_check():
+    failed = []
+    response = requests.get(prometheus + '/api/v1/query?', params={'query': 'ALERTS{alertname=~".*availability.*|.*requests.*|.*latency.*|.*response time.*", severity="critical"}'})
+    response_json = json.loads(response.text)
+    if len(response_json['data']['result']) < 1:
+        return failed
+    else:
+        for result in response_json['data']['result']:
+            failed.append("SLO - "+ result['metric']['sloth_slo'] + " is having active critical alert")
+        return failed
+
+def pods_details():
+    failed = []
+    failed_deploys = pod_check()
+    if len(failed_deploys) > 0:
+        response = requests.get(prometheus + '/api/v1/query?', params={'query': pods_details_query})
+        response_json = json.loads(response.text)
+        if len(response_json['data']['result']) < 1:
+            return failed
+        else:
+            for deploy in failed_deploys:
+                pods = []
+                for result in response_json['data']['result']:
+                    if str(result['metric']['pod']).startswith(str(deploy)):
+                        pods.append(str(result['metric']['pod']))
+                failed.append("Deployment - "+ str(deploy) +" does not have "+ healthy_pods +"% pods ready, failing pods are "+str(pods))
+            return failed
+    else:
+        return failed
+
+
+def main():
+    errors = []
+    for failures in [pods_details(), pvc_check(), node_check(), slo_check()]:
+        for error in failures:
+            errors.append(error)
+    print(errors)

```

Comment:
Remove this

Reply:
Removed print statement and added logger